rust/main/chains/hyperlane-sealevel/src/transaction.rs

```diff
@@ -0,0 +1,136 @@
+use std::collections::HashMap;
+
+use solana_sdk::pubkey::Pubkey;
+use solana_transaction_status::option_serializer::OptionSerializer;
+use solana_transaction_status::{
+    EncodedTransaction, EncodedTransactionWithStatusMeta, UiCompiledInstruction, UiInstruction,
+    UiMessage,
+};
+
+use hyperlane_core::H512;
+use hyperlane_sealevel_mailbox::instruction::Instruction;
+
+use crate::utils::{decode_h512, from_base58};
+
+/// This function searches for a transaction which dispatches Hyperlane message and returns
+/// list of hashes of such transactions.
+///
+/// This function takes the mailbox program identifier and the identifier for PDA for storing
+/// a dispatched message and searches a message dispatch transaction in a list of transaction.
+/// The list of transaction is usually comes from a block. The function returns list of hashes
+/// of such transactions.
+///
+/// The transaction will be searched with the following criteria:
+///     1. Transaction contains Mailbox program id in the list of accounts.
+///     2. Transaction contains dispatched message PDA in the list of accounts.
+///     3. Transaction is performing message dispatch (OutboxDispatch).
+///
+/// * `mailbox_program_id` - Identifier of Mailbox program
+/// * `message_storage_pda_pubkey` - Identifier for dispatch message store PDA
+/// * `transactions` - List of transactions
+pub fn search_transactions(
+    mailbox_program_id: &Pubkey,
+    message_storage_pda_pubkey: &Pubkey,
+    transactions: Vec<EncodedTransactionWithStatusMeta>,
+) -> Vec<H512> {
+    transactions
+        .into_iter()
+        .filter_map(|tx| match (tx.transaction, tx.meta) {
+            // We support only transactions encoded as JSON
+            // We need none-empty metadata as well
+            (EncodedTransaction::Json(t), Some(m)) => Some((t, m)),
+            _ => None,
```

Comment:
Ideally, no, since we choose the encoding in the RPC call. If we received a response which is not encoded correctly, the function will return an empty vector, and processing of dispatched messages will fail.

Reply:
could log a warning / error if it's not json so it's easier to keep track of; also in the other places where we `return None` because we don't support all formats yet (if we expect to miss anything currently)

Reply:
Added